src/client/components/Marketplace.tsx

```diff
@@ -1,16 +1,61 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import Page from './Page';
 import PageItem from './PageItem';
 import StatusPaper from './StatusPaper';
 import { Typography, Box } from '@mui/material';
 import { Marketplace as FusebitMarketplace } from '@fusebit/react-marketplace';
+import { useCustomColorsContext } from './useCustomColorsContext';
+import tinycolor from 'tinycolor2';
 
 const Marketplace = (props: {
   userData: UserData;
   onUninstall: (integrationId: string) => Promise<void>;
   getInstallUrl: (integrationId: string) => Promise<string>;
   isLoadingIntegrations?: boolean;
+  isInstalled?: boolean;
 }) => {
+  const { colors, isUsingCustomColors } = useCustomColorsContext();
+  const mainColor = colors.primary !== '#ffffff' ? colors.primary : colors.secondary;
+
+  useEffect(() => {
+    if (!props.isLoadingIntegrations && isUsingCustomColors) {
+      const interval = setInterval(() => {
+        const tile = document.querySelector('.tile') as HTMLElement;
+        const topContent = document.querySelector('.tile-top-content') as HTMLElement;
+        const title = document.querySelector('.tile-title') as HTMLElement;
+        const subtitle = document.querySelector('.tile-subtitle') as HTMLElement;
+        const button = document.querySelector('.tile-button') as HTMLElement;
+        const link = document.querySelector('.tile-link') as HTMLElement;
+        if (tile && topContent && tile && button && link && title && subtitle) {
+          const isDark = tinycolor(mainColor).isDark();
+          const lightTextColor = '#ffffff';
+          const darkTextColor = '#333333';
+
+          tile.style.boxShadow = 'none';
+          tile.style.fontFamily = 'Source Sans Pro';
+          tile.style.border = `1px solid ${tinycolor(mainColor).setAlpha(0.8).toRgbString()}`;
+          topContent.style.background = isDark
+            ? tinycolor(mainColor).lighten(18).setAlpha(0.2).toRgbString()
+            : tinycolor(mainColor).setAlpha(0.4).toRgbString();
+          title.style.color = isDark ? mainColor : darkTextColor;
+          title.style.fontWeight = '600';
+          subtitle.style.color = isDark
+            ? tinycolor(mainColor).darken(15).setAlpha(0.4).toRgbString()
+            : tinycolor(darkTextColor).lighten(25).toString();
+          console.log(button);
+          button.style.background = !props.isInstalled && mainColor;
+          button.style.color = isDark ? lightTextColor : darkTextColor;
+          button.style.fontWeight = '600';
+          link.style.borderColor = mainColor;
+          link.style.color = isDark ? mainColor : darkTextColor;
+          link.style.fontWeight = '600';
+
+          clearInterval(interval);
+        }
+      }, 50);
+    }
+  }, [colors, isUsingCustomColors, props.isLoadingIntegrations]);
```

Comment:
@fperaltalitebox I think we can change this by using props directly in the tile component

Reply:
Let's move this to a custom hook called useTitleContrastColor()